packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,143 @@
+import path from 'path';
+import { promisify } from 'util';
+import { Stream } from 'stream';
+import assert from 'node:assert';
+
+import { BuildFunction, BuildStepOutput } from '@expo/steps';
+import fs from 'fs-extra';
+import { Platform } from '@expo/eas-build-job';
+import fetch from 'node-fetch';
+import { repackAppAndroidAsync, repackAppIosAsync } from '@expo/repack-app';
+import { v4 as uuidv4 } from 'uuid';
+
+import IosCredentialsManager from '../utils/ios/credentials/manager';
+
+const pipeline = promisify(Stream.pipeline);
+
+export function createRepackBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: '__download_and_repack_golden_development_client_archive',
+    name: 'Download and repack golden development client archive',
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'repacked_archive_path',
+        required: true,
+      }),
+    ],
+    fn: async (stepsCtx, { outputs, env }) => {
+      const tmpDir = `/tmp/eas-build-golden-dev-client-app-${uuidv4()}`;
+      await fs.mkdirs(tmpDir);
+      stepsCtx.logger.info(`Created temporary directory: ${tmpDir}`);
+
+      stepsCtx.logger.info('Downloading golden development client archive...');
+      const fileName =
+        stepsCtx.global.staticContext.job.platform === Platform.IOS
+          ? 'golden-dev-client-latest.ipa'
+          : 'golden-dev-client-latest.apk';
+      const goldenArchiveUrl =
+        env.__EAS_GOLDEN_DEV_CLIENT_URL ??
+        `https://storage.googleapis.com/turtle-v2/onboarding/${fileName}`;
+      const goldenArchivePath = path.join(tmpDir, fileName);
+      try {
+        const response = await fetch(goldenArchiveUrl, {
+          timeout: 5 * 60 * 1000, // 5 minutes
```

Comment:
Why so huge? How big are the archives? Maybe we should have lesser timeout and retry?

Reply:
I made it 1 min